package/MDAnalysis/topology/PDBParser.py

```diff
@@ -96,64 +96,63 @@ def _parseatoms(self):
         iatom = 0
         atoms = []
 
-        with openany(self.filename, 'rt') as f:
-            resid_prev = 0  # resid looping hack
-            for i, line in enumerate(f):
-                line = line.strip()  # Remove extra spaces
-                if not line:  # Skip line if empty
-                    continue
-                record = line[:6].strip()
-
-                if record.startswith('END'):
-                    break
-                elif line[:6] in ('ATOM  ', 'HETATM'):
-                    try:
-                        serial = int(line[6:11])
-                    except ValueError:
-                        # serial can become '***' when they get too high
-                        self._wrapped_serials = True
-                        serial = None
-                    name = line[12:16].strip()
-                    altLoc = line[16:17].strip()
-                    resName = line[17:21].strip()
-                    # empty chainID is a single space ' '!
-                    chainID = line[21:22].strip()
-                    if self.format == "XPDB":  # fugly but keeps code DRY
-                        # extended non-standard format used by VMD
-                        resSeq = int(line[22:27])
-                        resid = resSeq
-                    else:
-                        resSeq = int(line[22:26])
-                        resid = resSeq
-
-                        while resid - resid_prev < -5000:
-                            resid += 10000
-                        resid_prev = resid
-                        # insertCode = _c(27, 27, str)  # not used
-                        # occupancy = float(line[54:60])
-                    try:
-                        tempFactor = float(line[60:66])
-                    except ValueError:
-                        tempFactor = 0.0
-                    segID = line[66:76].strip()
-                    element = line[76:78].strip()
-
-                    segid = segID.strip() or chainID.strip() or "SYSTEM"
-
-                    elem = guess_atom_element(name)
-
-                    atomtype = element or elem
-                    mass = get_atom_mass(elem)
-                    # charge = guess_atom_charge(name)
-                    charge = 0.0
-
-                    atom = Atom(iatom, name, atomtype, resName, resid,
-                                segid, mass, charge,
-                                bfactor=tempFactor, serial=serial,
-                                altLoc=altLoc, universe=self._u,
-                                resnum=resSeq)
-                    iatom += 1
-                    atoms.append(atom)
+        resid_prev = 0  # resid looping hack
+        for line in anyopen(self.filename, 'rt'):
```

Comment:
Ah, but maybe the `with` was more safe after reading about it.. I'll revert this